retroarch.c

```diff
@@ -30810,8 +30810,7 @@ static void video_driver_lock_new(struct rarch_state *p_rarch)
 void video_driver_set_cached_frame_ptr(const void *data)
 {
    struct rarch_state *p_rarch = &rarch_st;
-   if (data)
-      p_rarch->frame_cache_data = data;
+   p_rarch->frame_cache_data = data;
```

Comment:
I was a little unclear on the purpose of this function - as best I can tell, it's only used to clear out the frame pointer in scenarios such as unloading the core, and called with a null parameter, but the null check here prevents it from actually doing anything. If this essentially noop behavior is desired, I think we'll need a separate function to clear out the cached pointer from the driver side.

Reply:
It's more that this function was doing nothing before. I was considered something else might be relying on that, but all the call sites I could see looked like they were trying to clear it. But I don't know the source well :)